scripts/smoke_test.sh

```diff
@@ -0,0 +1,258 @@
+#!/bin/bash
+
+if [[ $1 == "" ]]; then
+  echo "Missing test image!"
+  exit 1
+fi
+
+image_name=$(basename $1)
+echo "=== Smoke testing ${image_name}"
+
+[[ $image_name =~ ^(.+)-(libvirt|ovirt)-([0-9]*)([a-z]*)(.+)\.qc2$ ]]
+product_name=${BASH_REMATCH[1]}              #manageiq
+platform=${BASH_REMATCH[2]}                  #libvirt
+product_version_number=${BASH_REMATCH[3]}    #18? (this is almost always blank, will fill in later)
+product_version_name=${BASH_REMATCH[4]}      #quinteros?
+product_version_remainder=${BASH_REMATCH[5]} #-1-20240403
+# Fill in product version number if missing
+if [[ $product_version_number == "" ]]; then
```

Comment:
Similarly here

```suggestion
if [[ "$product_version_number" == "" ]]; then
```

Reply:
also, not sure if `empty string` is more readable for you:

```
$ help test
[...]
    String operators:

      -z STRING      True if string is empty.

      -n STRING
         STRING      True if string is not empty.

      STRING1 = STRING2
                     True if the strings are equal.
      STRING1 != STRING2
                     True if the strings are not equal.
```

Reply:
Oh yeah good point.  I would use -z here possibly.